update-data.py

```diff
@@ -105,9 +77,33 @@ def insert_people(cursor, people):
                        ]
                       )
         for stint in person['stints']:
-            logging.debug("  Stint: {}, batch {}, {} - {}".format(
+            batch_id = None
+            batch = stint.get('batch')
+
+            if (batch):
+                batch_id = batch.get('id')
+
+                if (batch_id not in processed_batches):
+                    logging.debug("Batch {}, \"{}\", {} - {}".format(
+                        batch_id,
+                        batch.get('name'),
+                        stint.get('start_date'),
+                        stint.get('end_date')
+                    ))
+                    cursor.execute("INSERT INTO batches " +
+                                   " (batch_id, name, start_date, end_date)" +
+                                   " VALUES (%s, %s, %s, %s)",
+                                   [batch_id,
+                                    batch.get('name'),
+                                    stint.get('start_date'),
+                                    stint.get('end_date')
```

Comment:
I think this won't work quite right, as some people attend for a half batch and some attend for a full batch, and we'll store whichever we encounter first.

See, for example, the results of this query:

```sql
select batch_id, name, end_date - start_date from batches order by 1;
```

That said, it doesn't really matter, because we never actually look at `batches.start_date` or `batches.end_date`. More precise data lives on `stints`, so I think we can probably just drop the `batches.start_date` or `batches.end_date` columns entirely. That can happen later, though!

Reply:
Ah, I didn't account for the half-batches! I pushed another update that checks for `stint.for_half_batch = 'true'` before inserting the batches, and re-running your query, the length of batches seems to have been fixed. Can you double-check before I do the merge?

Reply:
I also just added a check to make sure the current person isn't faculty - they get stints with batch objects to set their `start_date` but the batch_id is `null`.

Reply:
I think they only have batch objects on the old API, which we're not using - I wrote more about that in a comment on the line with the check.

Reply:
I was still using the old APIs 🤦‍♀️! The latest commits reflect our last pairing session and should resolve this.